src/main/java/io/fabric8/jenkins/openshiftsync/PodTemplateUtils.java

```diff
@@ -412,4 +442,31 @@ protected static boolean hasSlaveLabelOrAnnotation(Map<String, String> map) {
                   && map.get("role").equals("jenkins-slave");
       return false;
   }
+
+
+  protected synchronized static void propogateProxyConfigToReservedPodTemplates(){
+    // Propogate Proxy config to node and maven podTemplates which cannot be changed otherwise
+    KubernetesCloud kubeCloud = JenkinsUtils.getKubernetesCloud();
+    LOGGER.info("Propogate proxy config to Reserved PodTemplates");
+    if (kubeCloud != null) {
+      List<PodTemplate> list = kubeCloud.getTemplates();
+      Iterator<PodTemplate> iter = list.iterator();
+      while (iter.hasNext()) {
```

Comment:
Maybe more readable to use a `for( PodTemplate pt : list )`  loop

Reply:
This implementation is fine I think as we would be makin changes to the list while reading the list. Probably would get a concurrency based exception.